client/src/components/SessionLock.vue

```diff
@@ -0,0 +1,74 @@
+<script>
+import { mapState, mapMutations } from 'vuex';
+
+export default {
+  name: 'SessionLock',
+  inject: ['user', 'djangoRest'],
+  computed: {
+    ...mapState(['mainSession']),
+  },
+  methods: {
+    ...mapMutations(['setMainSession']),
+    async acquireLock() {
+      try {
+        this.djangoRest.acquireSession(this.mainSession.id);
+        this.setMainSession({ ...this.mainSession, lock_owner: this.user });
+      } catch (err) {
+        console.log(err);
```

Comment:
We'll want to handle this more gracefully, as it's a situation that we should reasonably expect to occur -- two users trying to lock the session at roughly the same time.

Reply:
One idea I had was to just show a snack bar with message "This session is already owned" upon status 409 (i.e. someone else has acquired the lock in the time that the user was active and the endpoint raised `LockContention`). How should we update the local state of the lock owner when this occurs? Could we update `LockContention` to include the current lock owner or should we make another request to `/sessions` in order to update the local state?

Reply:
I think ultimately we'll want to hit an endpoint that refreshes the lock_owner. The way we've got the API set up now, retrieving the session is really expensive, so I think we'll need to either have a different endpoint for just the lock owner, or else move our existing expensive session retrieval endpoint. I'm leaning toward the latter, maybe move that from `GET /sessions/{id}` to `GET /sessions/{id}/deep` or something, so that a shallow lookup can be performed cheaply whenever we want.